### PR TITLE
Centralized interactive input() function that checks if STDIN is a TTY

### PR DESCRIPTION
**Pull request by derekbekoe**
- Promote _prompting to prompting
- Use prompt() to check TTY before getting user input

Closes https://github.com/Azure/azure-cli/issues/1349

### CODE REVIEW

**Comment by tjprescott** [comment]
Can we have a wrapper on getpass that confirms the typed password? I'd rather do that than roll my own in VM/VMSS create.

**Comment by derekbekoe** [comment]
User should always type the password twice?

File: src/azure-cli-core/azure/cli/core/commands/__init__.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import azure.cli.core.telemetry as telemetry
 from azure.cli.core._util import CLIError
 from azure.cli.core.application import APPLICATION
-from azure.cli.core._prompting import prompt_y_n, NoTTYException
+from azure.cli.core.prompting import prompt_y_n, NoTTYException
```
Comment (troydai) [inline src/azure-cli-core/azure/cli/core/commands/__init__.py]: Thank you!

File: src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py [hunk 2]
```diff
@@ -6,15 +6,15 @@
 from __future__ import print_function
 import os
 import sys
-from six.moves import input, configparser #pylint: disable=redefined-builtin
+from six.moves import configparser #pylint: disable=redefined-builtin
```
Comment (troydai) [inline src/command_modules/azure-cli-configure/azure/cli/command_modules/configure/custom.py]: PEP8: two spaces between the '#' and the previous char, one space between the '#' and the following char.

**Comment by tjprescott** [comment]
I would argue that yes, the user should always enter the password twice, but you could parameterize it with a default of False. The idea is that 1) the portal and like 99% of regular applications do this and 2) it would be very annoying to mistype your password and find you have no ability to access the VM you just waited to create because of a password typo. This at least ensures you got your password right or made the same mistake twice (in which case, no one can help you!)

**Comment by derekbekoe** [comment]
@tjprescott Added `prompt_pass` with confirmation option.

**Comment by tjprescott** [comment]
Thanks @derekbekoe !

**Pull request merged**